python/pyspark/sql/connect/dataframe.py

```diff
@@ -1836,10 +1839,21 @@ def _schema(self) -> StructType:
         if self._cached_schema is None:
             query = self._plan.to_proto(self._session.client)
             self._cached_schema = self._session.client.schema(query)
+            try:
+                self._cached_schema_serialized = CPickleSerializer().dumps(self._schema)
+            except Exception as e:
+                logger.warn(f"DataFrame schema pickle dumps failed with exception: {e}.")
+                self._cached_schema_serialized = None
         return self._cached_schema
 
     @property
     def schema(self) -> StructType:
+        if self._cached_schema_serialized is not None:
```

Comment:
I have a small nit. The first call will go through the deepcopy code path, the second call will use pickle. In practice this might be more of an issue for testing then for anything else.